src/videojs-hls.js

```diff
@@ -930,7 +951,11 @@ videojs.HlsHandler.prototype.loadSegment = function(segmentInfo) {
   this.segmentXhr_ = videojs.Hls.xhr({
     uri: segmentInfo.uri,
     responseType: 'arraybuffer',
-    withCredentials: this.source_.withCredentials
+    withCredentials: this.source_.withCredentials,
+    // Set xhr timeout to 150% of the segment duration to allow us
+    // some time to switch renditions in the event of a catastrophic
+    // decrease in network performance or a server issue.
+    timeout: (segment.duration * 1.5) * 1000
```

Comment:
Nice. We shouldn't assume `segment.duration` is always defined, though-- how about a fallback to `targetDuration`?